### PR TITLE
Revert to v1 registrations if not using WCA registration

### PR DESCRIPTION
**Pull request by dunkOnIT**
This will become unnecessary once we move to v3, but for now the default to v2 makes it impossible to import registrations for the comps not using WCA registrations

Have tested locally and can confirm:
- If "Use WCA registration" is ticked, it uses v2 registrations
- If "Use WCA registration" is NOT ticked, it uses v1 registratiosn

### CODE REVIEW

File: app/models/competition.rb [hunk 1]
```diff
@@ -2548,6 +2548,8 @@ def set_form_data(form_data, current_user)
       end
 
       assign_attributes(Competition.form_data_to_attributes(form_data))
+
+      self.uses_v2_registrations = false unless form_data.dig('website', 'usesWcaRegistration')
```
Comment (gregorbg) [inline app/models/competition.rb:2552]: At this point, the attributes from the form data have already been assigned (in the line directly above), so you might as well do `unless self.use_wca_registration`.

**Review by gregorbg — COMMENTED** [review]
Hmm... You definitely earn a "short and sweet" prize for coming up with this idea (and extra kudos for finding this place in the code to begin with, the WCAT form loading is a horrible mess!)

However I'm not convinced that it's the most clean place to implement this. In particular, it causes unintuitive side-effects for people in the UI. If you want to dabble with some React code, look at (for example) `FormSections/CompetitorLimit` to see how we enable/disable certain sections based on other parts of the form. It would definitely be better for the users to straight-out disable this in the UI already, to minimize confusion (otherwise, it would make it seem to someone who is explicitly turning on the checkbox in the UI that "the form isn't saving properly" because during the proper save, the checkbox value is being overridden without clear communication to the user)

if you don't have the time or motivation to look into this, let me know and I'll hijack the PR.

**Comment by dunkOnIT** [comment]
> otherwise, it would make it seem to someone who is explicitly turning on the checkbox in the UI that "the form isn't saving properly"

This concern isn't quite clicking with me? As I understand it, the only people who we expose the "enable v2" checkbox to are WCAT - and we can just "manually" communicate with them about this change. 

In the case of a normal user creating a competition - they would just uncheck the "Use WCA registration" checkbox and without them being aware of it, the default uses_v2_registration value is overridden. I don't see this as a concern/see how it would lead to confusion?

File: app/models/competition.rb (hunk 1 above)
Comment (dunkOnIT) [reply app/models/competition.rb]: Right you are!

**Comment by gregorbg** [comment]
This needs to be updated now that #10132 is merged.

> This concern isn't quite clicking with me? As I understand it, the only people who we expose the "enable v2" checkbox to are WCAT - and we can just "manually" communicate with them about this change.
>
> In the case of a normal user creating a competition - they would just uncheck the "Use WCA registration" checkbox and without them being aware of it, the default uses_v2_registration value is overridden. I don't see this as a concern/see how it would lead to confusion?

I will accept this explanation only if you add a comment stating that this line is a V2 hack that can be removed after full V3 rollout :smile:

**Review by gregorbg — APPROVED** [review]

**Pull request merged**